framework/sandstone.cpp

```diff
@@ -3736,8 +3822,9 @@ int main(int argc, char **argv)
     bool restarting = true;
     int total_tests_run = 0;
     TestResult lastTestResult = TestResult::Skipped;
+    auto it = test_set->begin();
 
-    for (struct test *test = get_next_test(tc); test; test = get_next_test(tc)) {
+    for (struct test *test = get_next_test(it, tc); test; test = get_next_test(it, tc)) {
```

Comment:
And if you use the struct return type I suggested above, this would look like, here:
```c++
    for (auto next = get_next_test(test_set->begin(), tc); next.test; next = get_next_test(next.it, tc))
```

An alternative is to create a Java iterator-like struct:
```c++
    struct TestIteration iteration(test_set);
    while (iteration.hasNext()) {
        struct test *test = iteration.test;
        int tc = iteration.tc;
        ....
```

Reply:
I see changes in later commits that may obviate this. Would be nice if you said in the commit message if this is temporary.